core/domain/collection_services.py

```diff
@@ -188,8 +238,21 @@ def get_collection_summary_by_id(collection_id):
 
 def get_multiple_collections_by_id(collection_ids, strict=True):
     """Returns a dict of domain objects representing collections with the
-    given ids as keys. If a collection_id is not present it is not included in
-    the return dict.
+    given ids as keys.
+
+    Args:
+        collection_ids: list(str). A list of collection ids of collections to
+            be retrieved.
+        strict: bool. Whether to fail noisily if no
+            collection with a given id exists in the datastore.
+
+    Returns:
+        A dict of domain objects representing collections with the given ids as
+        keys.
+
+    Raises:
+        ValueError: Only if strict is True. One or more collections with the
```

Comment:
ValueError: 'strict' is True, and one or more of the given collection ids are invalid.

(Rationale for latter part of change: "collection cannot be found" feels a bit off, it implies that a collection exists but is somehow hidden.)